### PR TITLE
vcf_summary: Fix basic checklist items for DX apps

### PR DESCRIPTION
**Pull request by msimbirsky**
- dxapp.json: Add input help text, citation stub, regions, timeout, distro
- Note an unimplemented option
- Consistent whitespace/indentation in the main script
- Remove dead code
- Remove boilerplate comments from Readme.md and main script
- Set shell options '-e -o pipefail'

### CODE REVIEW

**Comment by msimbirsky** [comment]
#dxapp.json
- [x] Upgrade to 14.04
- [x] Add help and extension to input spec.
- [x] In `details` field specify: `whatsNew`and `citation` (leave blank for citation.
- [x] In `details` consider specifying `upstreamAuthor` and `upstreamVersion` fields when applicable.
- [x] Use `regionalOptions` 
- [x] Set applicable timeout (Be relax) and restart policy in [`runSpec`](https://wiki.dnanexus.com/API-Specification-v1.0.0/IO-and-Run-Specifications).
- [x] Add subjob restartability when applicable.
- [x] `version` field, set proper [semantic versioning](https://semver.org/).
- [x] Setting `"openSource": true`. 
- [ ] Run [dxapp linter](https://github.com/dnanexus/RoughER/blob/master/scripts/dxapp_lint.py) [[permalink](https://github.com/dnanexus/RoughER/blob/ecb3ea3f97870c020f93bf4723afc9f40a310e33/scripts/dxapp_lint.py)] on `dxapp.json`.
- [x] Use AssetDepend instead of execDepend
- [x] Add default instance type that works for a general use case.
- [x] add new input if applicable

#src/code.sh src/code.py
- [ ] GATK tarball should be removed from the resource project (or make a new resource project), and the applet should require GATK tarball as a required input. This way, the users are enforced to provide their own tarball. This would avoid trouble from non-academic entities without GATK license to use your tarball.

# Readme.md
- [x] Improve Readme file to contain applet function, use case, input/output description, underline execution if it’s a wrapper.
- [x] Read documentation for sanity check. (Unpublished app URL will be: https://platform.dnanexus.com/app/app-xxxx)

# Test
- [ ] Basic testing using given input in Box or expected input
- [ ] Make a Pull Request to `master` in https://github.com/RitchieLab/dnanexus

# Production
- [ ] Co-test with Ritchie group
- [ ] Help Ritchie group to publish (Give dx command)
- [ ] Verify the versions that appear on the platform are the intended versions
- [ ] Set up demo workflow in public project.

**Comment by msimbirsky** [comment]
Missing "variant-level summary" script

File: sequencing/vcf_summary/dxapp.json [hunk 1]
```diff
@@ -3,34 +3,42 @@
...
   "categories": [
     "Statistics"
   ],
+  "details": {
+    "whatsNew": "",
+    "citation": ""
+  },
   "inputSpec": [
     {
       "name": "vcf_fn",
```
Comment (slzarate) [inline sequencing/vcf_summary/dxapp.json]: Change to `variants_vcfgzs`

File: sequencing/vcf_summary/dxapp.json [hunk 2]
```diff
@@ -3,34 +3,42 @@
...
-      "label": "VCF File",
+      "label": "VCF file",
```
Comment (slzarate) [inline sequencing/vcf_summary/dxapp.json]: I think we decided on capitalizing all words in the label

File: sequencing/vcf_summary/Readme.md [hunk 3]
```diff
@@ -1,21 +1,6 @@
 <!-- dx-header -->
 # VCF Summary Statistics (DNAnexus Platform App)
 
-Generate summary statistics for a VCF file (by sample and by variant)
+Generate summary statistics for a VCF file (by sample and by variant).
```
Comment (slzarate) [inline sequencing/vcf_summary/Readme.md]: to write

File: sequencing/vcf_summary/dxapp.json [hunk 4]
```diff
@@ -3,34 +3,42 @@
   "title": "VCF Summary Statistics",
   "summary": "Generate summary statistics for a VCF file (by sample and by variant)",
   "dxapi": "1.0.0",
-  "version": "0.0.1",
+  "version": "0.1.0",
```
Comment (etal) [inline sequencing/vcf_summary/dxapp.json]: I think we're setting the versions to 1.0.0, not 0.1.0. (I had it wrong initially.)

**Comment by msimbirsky** [comment]
@etal @slzarate Review amendments are done

**Pull request merged**